game/theater/start_generator.py

```diff
@@ -350,7 +356,13 @@ def generate_base_defenses(self) -> None:
                 self.generate_garrison()
 
     def generate_garrison(self) -> None:
-        position = self._find_location()
+        position = self.control_point.preset_locations.random_garrison()
+        if position is None:
+            logging.warning(
+                f"Found no preset location for {self.control_point} garrison. "
+                "Falling back to random location."
+            )
+            position = self._find_random_location()
```

Comment:
All of these could probably be be piped through a `preset_exists` or similar method rather than being repeated for each type.